### DISCUSSION TITLE
"Is Unique" Toggle is Hidden on Column that is Part of a Table's Composite Primary Key

### DISCUSSION THREAD

**Discussion by raparri01** [supabase/supabase#16911] [Contribute to Supabase]
# Bug report

## Describe the bug

When a column is designated as a Primary Key, the "Is Unique" flag does not appear in the Table Editor settings. This behavior makes sense when a single column is the Primary key, but is frustrating to deal with when you are using a composite primary key where a unique combination of 2 or more columns is the primary key.

Since the "Is Unique" flag is hidden, you can run into a uniqueness violations when writing to the table and not know why. 

## To Reproduce

1. Go to the Table Editor
2. Designate 2 columns as Primary Keys
3. Attempt to edit a column's "Is Unique" status

## Expected behavior

1. Go to the Table Editor
2. Designate 2 columns as Primary Keys
3. Attempt to edit a columns "Is Unique" status
4. See the columns "is unique" status

## System information
Web Table Editor

**Comment by joshenlim** [comment]
@MildTomato we'll need to keep this in mind when we get to relooking at the UI of the table editor
Not just composite primary keys, but foreign keys as well

**Comment by aayushdura** [comment]
@raparri01 @joshenlim  So in Front-end currently there is this key icon to indicate the column as a primary key.
![image](https://user-images.githubusercontent.com/58044533/175901858-14790799-7246-4fa8-9f8a-04d37a6ce6f5.png)
Currently if we assign a column as isUnique then there's no indication to the respective column,
Can I add a identifier to the column which will denote the column as isUnique or not

**Comment by joshenlim** [comment]
Hey @aayushdura! Thanks so much for this - but this doesnt solve the issue that this ticket is mentioning though.

Atm our table grid doesnt support composite column stuffs that well (e.g foreign keys and in this context, setting "unique" across composite columns)

On the topic of having an icon to denote the unique property of the column on the other hand, I personally dont think we should as it'll add clutter to the table editor (pertaining to UX). As much as possible we should only be showing the important information at the first level, all else could be found on the second level. (Cc @MildTomato if you might have opinions here too)

For your case specifically you can check if a column has is unique toggled by right clicking on the column and editing it. You'll see a checkbox there which denotes is the column is unique or not, and you can edit it accordingly too.

**Comment by sonu27** [comment]
I just ran into this issue. Had to delete the table and re-add, which is less than ideal. I'm sure SQL could be ran to fix, but the UI blocks removing the contraint

**Comment by silentworks** [comment]
Anyone hitting this issue, you can use the SQL editor for now if you wish to remove a constraint from a table.

```sql
ALTER TABLE table_name
DROP CONSTRAINT constraint_name;
```

If you don't know the constraint name, you can run another query to get all constraints on that table:
```sql
SELECT con.*
       FROM pg_catalog.pg_constraint con
            INNER JOIN pg_catalog.pg_class rel
                       ON rel.oid = con.conrelid
            INNER JOIN pg_catalog.pg_namespace nsp
                       ON nsp.oid = connamespace
       WHERE nsp.nspname = 'public'
             AND rel.relname = 'table name';
```
